lib/Tvm.js

```diff
@@ -273,6 +274,10 @@ class Tvm {
         await this._validateNamespaceInApprovedList(params.owNamespace, params.approvedList)
       } catch (e) {
         logger.warn(`unauthorized request: ${e.message}`)
+        if (e.code && !e.code.startsWith('4')) {
+          // return 500 error if its not a 4xx error
+          return Tvm._userErrorResponse('server error', 500)
```

Comment:
wdyt @sandeep-paliwal ?

Reply:
@moritzraho valid point, but only in case of a 429 error a retry will be relevant (that too if TVM client has logic to wait for the 'retry-after' header value, else we keep getting 429 rate limiting error for retries), for all other 4xx error which OW might throw retry call will fail. So I still feel the current logic to wrap all 4xx as 403 unauthorised works, and we don't need to look for specific 4xx.

Reply:
Ok maybe let's forget the retry for a bit, IMO it seems more robust to return a 500 if we got an unexpected error, as we got an error we could not handle. This will allow us to monitor unexpected errors, otherwise we might think everything is fine if TVM returns a 403

Reply:
yeah, it's not a big impact either ways. Will change to 500 error except for 400,401 and 403.

Reply:
ok thanks for the changes
What about if `e.code` is not defined, it should throw a 500 too right ?

Reply:
I think we can simply throw the error in line 140 without converting to string
then in line 283 just do `if (!e.statusCode || e.statusCode !== 403, ...or array.include)` ?

Reply:
I guess it's better to set 403 in `_validateNamespaceInApprovedList ` and keep remaining logic as is. With this there won't be case of undefined `e.code`